dpnp/linalg/dpnp_iface_linalg.py

```diff
@@ -752,6 +930,73 @@ def matmul(x1, x2, /):
     return dpnp.matmul(x1, x2)
 
 
+def matrix_norm(x, /, *, keepdims=False, ord="fro"):
+    """
+    Computes the matrix norm of a matrix (or a stack of matrices) `x`.
+
+    This function is Array API compatible.
+
+    For full documentation refer to :obj:`numpy.linalg.matrix_norm`.
+
+    Parameters
+    ----------
+    x : {dpnp.ndarray, usm_ndarray}
+        Input array having shape (..., M, N) and whose two innermost
+        dimensions form ``MxN`` matrices.
+    keepdims : bool, optional
+        If this is set to ``True``, the axes which are normed over are left in
+        the result as dimensions with size one. With this option the result
+        will broadcast correctly against the original `x`.
+        Default: ``False``.
+    ord : {None, 1, -1, 2, -2, dpnp.inf, -dpnp.inf, 'fro', 'nuc'}, optional
```

Comment:
Should we list `int, float` here also? At least from the array API:

> ord: int | float | ~typing.Literal[inf, -inf, 'fro', 'nuc'] | None = 'fro'

Reply:
I do not think so. It seems a mistake in Array API. Look at the table given after `ord` definition in [vector_norm](https://data-apis.org/array-api/latest/extensions/generated/array_api.linalg.vector_norm.html#array_api.linalg.vector_norm), there are specific rows for `(int,float >= 1)` and `(int,float < 1)`. While for [matrix_norm](https://data-apis.org/array-api/latest/extensions/generated/array_api.linalg.matrix_norm.html), there is nothing in the table. Similarly, in the Note section of [numpy.linalg.norm](https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.htm), for "other" values norm is only defined for vectors not matries.